src/analysis/problematic_exclusions.py

```diff
@@ -0,0 +1,370 @@
+    df_summary = pd.DataFrame([{
+        summary_count_field: summary_count,
+        'pct_in1_notIn2_in3__over_in1': round(summary_count / len(mirror_set), 4) if summary_count > 0 else 0
+    }])
+
+    # Format
+    df = df.sort_values(
+        [term_label_field, analytical_field, term_id_field, mirror_field, component_field, mappings_field],
+        ascending=[False, True, False, False, False, False])
+    df_summary = df_summary.sort_values([summary_count_field], ascending=[False])
+
+    # Save & return
+    if save:
+        df.to_csv(outpath, sep='\t', index=False)
+        df_summary.to_csv(outpath_xref_summary, sep='\t', index=False)
+
+    return {outpath: df, outpath_xref_summary: df_summary}
+
+
+def cli_validate(d: Dict) -> Dict:
+    """Validate CLI args. Also updates these args if/as necessary"""
+    # Remove unused args
+    d = {k: v for k, v in d.items() if v}
+
+    # Outpath
+    # - absolute path: Assumes relative paths are relative to /src/ontology/
+    # todo: Do this for all path args
+    if not os.path.isabs(d['outpath']):
+        d['outpath'] = os.path.join(ONTOLOGY_DIR, d['outpath'])
+
+    return d
+
+
+def cli() -> Dict[str, pd.DataFrame]:
+    """Command line interface."""
+    parser = cli_get_parser()
+    kwargs = parser.parse_args()
+    kwargs_dict: Dict = vars(kwargs)
+    kwargs_dict = cli_validate(kwargs_dict)
+    del kwargs_dict['onto_name']  # todo: if really don't need, remove this line & remove CLI param & from makefile
+    output_xrefs, output_classes = {}, {}
+    if 'excluded_terms_in_mondo_xrefs' in kwargs_dict['outpath']:
+        d = {k: v for k, v in kwargs_dict.items() if k != 'mondo_path'}
+        output_xrefs = run_excluded_terms_in_mondo_xrefs(**d)
+    if 'excluded_terms_in_mondo' in kwargs_dict['outpath']:
+        d = {k: v for k, v in kwargs_dict.items() if k != 'mondo_mappings_path'}
+        # todo: remove this noinspection when done
+        # noinspection PyArgumentList
+        output_classes = run_excluded_terms_in_mondo(**d)
+    outputs = {**output_xrefs, **output_classes}
+    return outputs
+
+
+def cli_get_parser() -> ArgumentParser:
+    """Add required fields to parser."""
+    package_description = \
+        'Determine which terms outside the term scope of the Mondo source alignment are xrefed in Mondo.'
+    parser = ArgumentParser(description=package_description)
+
+    parser.add_argument(
+        '-n', '--onto-name', required=True,
+        help='Name of the ontology.')
+    parser.add_argument(
+        '-O', '--onto-path', required=True,
+        help='Optional. Path to the ontology file to query.')
+    parser.add_argument(
+        '-c', '--onto-config-path', required=True,
+        help='Optional. Path to a config `.yml` for the ontology which contains a `base_prefix_map` which contains a '
+             'list of prefixes owned by the ontology. Used to filter out terms.')
+    # parser.add_argument(
```

Comment:
Is it better to use the exclusion tables as an input for this rather than component and mirror? I'm guessing not, but just bringing it up just in case.

I was thinking this would be the much easier thing to do; simply use the exclusion tables I just created. I got ahead of myself, and had to backtrack my refactoring after I realized that the nature of my outputs depended on those files.

Reply:
Its ok either way. mirror-components=exclusions, and the way you have it now has more information, so I prefer it.